content/3_Silver/Intro_Bitwise.mdx

```diff
@@ -377,118 +364,41 @@ Therefore, the binomial coefficient is odd if: $pref_n-pref_k-pref_{n-k}=0$.
 
 ```cpp
 #include <iostream>
+#include <numeric>
 #include <vector>
 
 using namespace std;
 
 int main() {
 	int n;
 	cin >> n;
-	vector<int> pref(n + 1);
-	for (int i = 2; i <= n; i++) {
-		int num = i;
-		while (num % 2 == 0) {
-			pref[i]++;
-			num /= 2;
-		}
-		pref[i] += pref[i - 1];
-	}
-
-	int ans = 0;
-	for (int i = 0; i < n; i++) {
-		int num;
-		cin >> num;
-		if (pref[n - 1] - pref[i] - pref[n - i - 1] == 0) { ans ^= num; }
-	}
-
-	cout << ans << endl;
-}
-```
-
-</CPPSection>
-
-<JavaSection>
-```java
-import java.io.*;
-import java.util.*;
-
-public class XorPyramid {
-	public static void main(String[] args) throws IOException {
-		BufferedReader f = new BufferedReader(new InputStreamReader(System.in));
-		PrintWriter out = new PrintWriter(System.out);
-		StringTokenizer st = new StringTokenizer(f.readLine());
-
-		int n = Integer.parseInt(st.nextToken());
-
-		// Precompute the powers of 2 in the factorization of i!
-		int[] pref = new int[n + 1];
-		for (int i = 2; i <= n; i++) {
-			int num = i;
-			while (num % 2 == 0) {
-				pref[i]++;
-				num /= 2;
-			}
-			pref[i] += pref[i - 1];
-		}
-
-		int ans = 0;
-		st = new StringTokenizer(f.readLine());
-		for (int i = 0; i < n; i++) {
-			int num = Integer.parseInt(st.nextToken());
-
+	vector<int> v(n);
+	for (int &a : v) { cin >> a; }
+	long long ans = -accumulate(v.begin(), v.end(), 0LL);
+	// For every bit position, check if it's set on in the xor-sum of every
+	// subsequence
+	for (int i = 0; i < 30; i++) {
+		int s = 0;
+		// Count the prefix sums
+		vector<int> pref = {1, 0};
```

Comment:
why does this start w/ a 1

Reply:
It's like the whole prefix.

Reply:
that...does not clear things up

Reply:
In the code I search prefixes $\texttt{pref}_j$ such that $\texttt{pref}_j \oplus \texttt{pref}_i = 1$. The $1$ accounts for when the xor-sum of the whole prefix is $1$, that otherwise wouldn't be counted.